dipy/viz/actor.py

```diff
@@ -1378,16 +1381,22 @@ def sphere(centers, colors, radii=1., theta=16, phi=16):
     radii_fa = numpy_support.numpy_to_vtk(radii.astype('f8'), deep=0)
     radii_fa.SetName('rad')
 
-    src = vtk.vtkSphereSource()
-    src.SetRadius(0.5)
-    src.SetThetaResolution(theta)
-    src.SetPhiResolution(phi)
-
     polyData = vtk.vtkPolyData()
-    polyData.SetPoints(pts)
-    polyData.GetPointData().AddArray(radii_fa)
-    polyData.GetPointData().SetActiveScalars('rad')
-    polyData.GetPointData().AddArray(cols)
+    if faces is None:
```

Comment:
Hi @thechargedneutron you will need to rebase this on top of https://github.com/nipy/dipy/pull/1528 (my many_spheres branch) and update the test too.